client/components/domain-search-v2/register-domain-step/index.jsx

```diff
@@ -743,13 +743,24 @@ class RegisterDomainStep extends Component {
 	}
 
 	renderFilterResetNotice() {
+		const { exactSldMatchesOnly = false, tlds = [] } = this.state.lastFilters;
+		const hasActiveFilters = exactSldMatchesOnly || tlds.length > 0;
+
+		const suggestions = this.state.searchResults;
+		const hasTooFewSuggestions = Array.isArray( suggestions ) && suggestions.length < 10;
+
+		if ( ! ( hasActiveFilters && hasTooFewSuggestions ) ) {
```

Comment:
This is a weird if. We can rewrite it as `!hasActiveFilters || !hasTooFewSuggestions` for clarity. But not a blocker.

Reply:
It was originally what you proposed but I decided it's not clear :D